src/mixins/segmentation/freehandSegmentationMixin.js

```diff
@@ -0,0 +1,189 @@
+import external from '../../externalModules.js';
+import store from '../../store';
+import { getLogger } from '../../util/logger.js';
+import { draw, drawJoinedLines, getNewContext } from '../../drawing';
+
+const logger = getLogger('tools:ScissorsTool');
+
+const { getters, setters } = store.modules.segmentation;
+
+/**
+ * Render hook: draws the FreehandScissors's outline
+ *
+ * @param {Object} evt Cornerstone.event#cornerstoneimagerendered > cornerstoneimagerendered event
+ * @memberof Tools.ScissorsTool
+ * @returns {void}
+ */
+function renderToolData(evt) {
+  const eventData = evt.detail;
+  const { element } = eventData;
+  const color = getters.brushColor(element, true);
+  const context = getNewContext(eventData.canvasContext.canvas);
+  const handles = this.handles;
+
+  draw(context, context => {
+    if (handles.points.length > 1) {
+      for (let j = 0; j < handles.points.length; j++) {
+        const lines = [...handles.points[j].lines];
+        const points = handles.points;
```

Comment:
`const points = handles.points;` can be shifted up to the top level, as you're redefining it for the same value on each iteration.

Then your checks become: `if(points.length)` and `const lines = [...points[j].lines];`